### PR TITLE
add response to cars, refactor open_alpr api logic

### PR DESCRIPTION
**Pull request by omarosama1989**

### CODE REVIEW

File: app/models/car.rb [hunk 1]
```diff
@@ -3,7 +3,32 @@ class Car < ApplicationRecord
...
+    end
+    self.update(attrs)
+
+
+  end
+
+  private
+  def serialize_response(response)
+    result = response["results"].first
+    attrs = unless result.nil?
```
Comment (aalyahya) [inline app/models/car.rb]: Never use (unless else), use (if else) instead.

You can refactor this to: if results.present?

File: app/controllers/cars_controller.rb [hunk 2]
```diff
@@ -15,30 +15,19 @@ def index
...
-      response = OpenAlprService.new.recognize_url(image_url: add_uri_to_url(@car.image.url), country: @car.country)
-      unless response.nil?
-        @car.update(response)
-        redirect_to action: :index
-      else
-        @car.update(found: false)
-        @car.errors.add(:image, "No result found")
-        render :index
-      end
+      @car.open_alpr_recogmize_url("#{request.protocol}#{request.host}")
```
Comment (aalyahya) [inline app/controllers/cars_controller.rb]: move this call to the model using after_create callback

File: app/models/car.rb [hunk 3]
```diff
@@ -3,7 +3,32 @@ class Car < ApplicationRecord
   mount_uploader :image, ImageUploader
 
   validates :image, :country, presence: true
-  def options_for_countries
 
+  def open_alpr_recogmize_url(domain)
+    response = OpenAlprService.new.recognize_url(image_url: "#{domain}/#{self.image.url}", country: self.country)
```
Comment (aalyahya) [inline app/models/car.rb]: * Make the URL static 
* Parse the response in the lib and then return an object to the model

File: app/models/car.rb [hunk 4]
```diff
@@ -3,7 +3,32 @@ class Car < ApplicationRecord
...
+    else
+      attrs = serialize_response(response)
+    end
+    self.update(attrs)
+
+
+  end
+
+  private
+  def serialize_response(response)
```
Comment (aalyahya) [inline app/models/car.rb]: move it to the service

File: lib/open_alpr_service.rb [hunk 5]
```diff
@@ -12,19 +12,12 @@ def recognize_url(image_url: nil, country: COUNTRY_CODE)
     unless image_url.nil?
       params = {image_url: image_url, secret_key: @secret_key, country: country}
```
Comment (aalyahya) [inline lib/open_alpr_service.rb]: move it to app/services

File: app/models/car.rb (hunk 1 above)
Comment (omarosama1989) [reply app/models/car.rb]: Is that because *else block* will be executed in most cases?

File: lib/open_alpr_service.rb (hunk 5 above)
Comment (omarosama1989) [reply lib/open_alpr_service.rb]: done

File: app/controllers/cars_controller.rb (hunk 2 above)
Comment (omarosama1989) [reply app/controllers/cars_controller.rb]: Done

File: app/models/car.rb (hunk 1 above)
Comment (omarosama1989) [reply app/models/car.rb]: Done

File: app/models/car.rb (hunk 3 above)
Comment (omarosama1989) [reply app/models/car.rb]: I left a comment yesterday but I think I missed clicking the button.

- I made the URL as a ```environment variable``` in `local_env_variables.yml`. Is that fine?
- I think that parsing the response to ***car*** model is not the ***open_alpr_service*** responsibility. It is the model's. Do you have other point of view I did not see?

**Pull request merged**